cmd/cli/plugin/secret/registry_secret_add.go

```diff
@@ -74,16 +84,40 @@ func registrySecretAdd(cmd *cobra.Command, args []string) error {
 			}
 		}
 		log.Info("\n")
-	}
-
-	pkgClient, err := tkgpackageclient.NewTKGPackageClient(registrySecretOp.KubeConfig)
-	if err != nil {
-		return err
-	}
+	} else {
+		secretExportList, err := pkgClient.ListSecretExports(registrySecretOp)
+		if err != nil {
+			return err
+		}
 
-	kc, err := kappclient.NewKappClient(registrySecretOp.KubeConfig)
-	if err != nil {
-		return err
+		for j := range secretExportList.Items {
+			export := ""
+			secretExport := secretExportList.Items[j]
+			if secretExport.Name == registrySecretOp.SecretName {
+				ns := &corev1.NamespaceList{}
+
+				err = kc.GetClient().List(context.Background(), ns)
+				if err != nil {
+					return err
+				}
+
+				if findInList(secretExport.Spec.ToNamespaces, "*") || secretExport.Spec.ToNamespace == "*" {
+					export = "all namespaces"
+				} else {
+					export = "some namespaces"
+				}
+
+				// Ask user consent when SecretExport has been created by kubectl and user tries to add secret of the same name as SecretExport without using --export-to-all-namespaces flag
+				log.Warningf("Warning: SecretExport with the same name exists already, given secret contents will be available to %s. If you decide not to proceed, you can either delete the SecretExport or specify a different secret name.\n\n", export)
+				if !registrySecretOp.SkipPrompt {
+					if err := cli.AskForConfirmation("Are you sure you want to proceed?"); err != nil {
+						return errors.New("creation of the secret got aborted")
+					}
+				}
+				log.Info("\n")
```

Comment:
Question: is there any specific reason we want to print a change-line via `log.Info` ? The reason why I am asking is that I see we already have two `\n`s from https://github.com/vmware-tanzu/tanzu-framework/pull/1088/files#diff-acb06893c75ae6caca902b2669b6d4d6fd949951490626d59d8241b0b03088b8R111

Reply:
I respond as I had added the same to the ```registry_secret_add.go``` and ```registry_secret_update.go```. That is to add a separation after the warning message and the rest of the flow, so for aesthetic reasons.

Reply:
Thanks for the explanation. There are many other places have `\n`s constructed within the message string. It might become hard to maintain with the growth of our code base. 

- If the newer log lib has some enhancements on `change-lines`, I as a developer might have a hard time to make modifications accordingly in all places and it also takes time to test all messages.
- I am also thinking about the "consistency" among all plugins. Other plugin owner might not append `\n` within their messages or they might think `\n\n\n` is better. We might be fine in this case as long as users are ok with it.

Based on above reasons(maintainability and consistency), I feel like building an API on top of `log.xxx` might be a good idea. Developers just need to provide the content of a message without worrying about `change-lines`. This is definitely something could be addressed in the future, not a blocker for this PR to get merged.